python/ray/train/_internal/dataset_spec.py

```diff
@@ -137,6 +137,8 @@ def preprocess_datasets(
 
             for key, dataset in datasets.items():
                 conf = self._config(key)
+                if conf.randomize_block_order:
+                    dataset = dataset.randomize_block_order()
```

Comment:
Should we add `randomize_block_order_each_window` for the pipelined case?

Reply:
Yeah, I was debating that. I guess we may as well in case it helps with randomization, even though it doesn't help with hotspots.